fgpyo/sam/__init__.py

```diff
@@ -180,8 +180,16 @@
 from pysam import AlignmentFile as SamFile
 from pysam import AlignmentHeader as SamHeader
 from pysam import qualitystring_to_array
-from typing_extensions import Self
-from typing_extensions import deprecated
+
+if sys.version_info > (3, 10):
```

Comment:
Added in 3.11: https://docs.python.org/3/library/typing.html#typing.Self

Reply:
Because `sys.version_info()` has a `micro` field it will be better to do:

```suggestion
if sys.version_info[:2] >= (3, 11):
```

For example, see the full tuple like this:

```python
>>> tuple(sys.version_info)
(3, 11, 7, 'final', 0)
```